sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -102,16 +105,24 @@ public void setTransaction(@Nullable String transaction) {
    * @return the user
    */
   public @Nullable User getUser() {
-    return user;
+    synchronized (userLock) {
+      if (user == null) {
```

Comment:
A couple of potential improvements:

* double-check lock
* account for when no user is ever set. Now it'll take the lock and check the cache on each use.

Reply:
I've noticed the multiple checks of `getUser` call.
Ideally, we'd only check it again if the value has been changed.
Definitely improvements to be done.